plugin/mouse/mouse.py

```diff
@@ -278,29 +294,33 @@ def show_cursor_helper(show):
 @ctx.action_class("user")
 class UserActions:
     def noise_trigger_pop():
-        if settings.get("user.mouse_enable_pop_stops_scroll") >= 1 and (
-            gaze_job or scroll_job
-        ):
-            # Allow pop to stop scroll
-            stop_scroll()
+        if actions.speech.enabled():
```

Comment:
Could we not use a context for this instead? Ie have the noise_trigger_pop action defined to this in wake mode and the double pop checker in sleep

Reply:
I have no clue what to do here - I did not wrap my head around context.

Reply:
@pokey 

Is that syntax available in non beta talon? This is how it's defined in your repo:

```py
mode: sleep
-
parrot(postalveolar_click): user.wake_all()
```

I'm not sure if that syntax is available for vanilla talon via `noise(pop)`

Reply:
I don't believe it is available, but you might ask in the talon slack workspace. Talon scripting channel should do

Reply:
Yes, it looks like using noise() is unavailable in .talon files, so as written this implementation seems like the best way to go about this currently.

Reply:
> Could we not use a context for this instead? Ie have the noise_trigger_pop action defined to this in wake mode and the double pop checker in sleep

I still think this would be better than the big if-else statement

Reply:
Yeah I was thinking about this as well, what would that format be instead of how it's structured now?

```py
# double_pop_trigger_wake.py

ctx.matches = r"""
mode: sleep
"""

@ctx.action_class("user")
class UserActions:
    def noise_trigger_pop():
        # code etc
```

Reply:
I don't see a point in having `pop_twice_to_wake` in a module class. Just have the function itself in the file. I would do something like this?

```py
from talon import Context, settings
import time

ctx = Context()

ctx.matches = r"""
mode: sleep
"""

time_last_pop = 0

def pop_twice_to_wake():
        """Use pop sound to wake from sleep"""
        global time_last_pop
        delta = time.time() - time_last_pop
        if delta >= 0.1 and delta <= 0.3:
            actions.speech.enable()
        time_last_pop = time.time()


@ctx.action_class("user")
class UserActions:
    def noise_trigger_pop():
        if settings.get("user.mouse_enable_pop_wake") == 1:
                pop_twice_to_wake()
```

Reply:
Nice, this works without having to nest the function within `mouse.py`

Reply:
Nice, this does feel clearer than having wake functionality nested inside `mouse.py`.

Reply:
thanks @AndreasArvidsson - I've updated the PR accordingly